src/vs/workbench/electron-browser/workbench.ts

```diff
@@ -343,7 +351,14 @@ export class Workbench extends Disposable implements IPartService {
 		serviceCollection.set(IListService, this.instantiationService.createInstance(ListService));
 
 		// Context Menu
-		serviceCollection.set(IContextMenuService, new SyncDescriptor(ContextMenuService));
+		const hasCustomTitle = this.getCustomTitleBarStyle() === 'custom';
+
+		// Use themable context menus when custom titlebar is enabled to match custom menubar
+		if (isWindows && hasCustomTitle) {
```

Comment:
@sbatten why not linux here? It seems to me your custom menu can work on Linux too?